network/retrieval/retrieve_test.go

```diff
@@ -137,6 +138,49 @@ func TestChunkDelivery(t *testing.T) {
 	}
 }
 
+// TestNoSuitablePeer brings up two nodes, tries to retrieve a chunk which is never
+// found, expecting a NoSuitablePeer error from netstore
+func TestNoSuitablePeer(t *testing.T) {
+	nodes := 2
+
+	sim := simulation.NewBzzInProc(map[string]simulation.ServiceFunc{
+		"bzz-retrieve": newBzzRetrieveWithLocalstore,
+	}, true)
+	defer sim.Close()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	_, err := sim.AddNodesAndConnectFull(nodes)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	result := sim.Run(ctx, func(ctx context.Context, sim *simulation.Simulation) error {
+		nodeIDs := sim.UpNodeIDs()
+		if len(nodeIDs) != nodes {
+			t.Fatal("not enough nodes up")
+		}
+		// allow the two nodes time to set up the protocols otherwise kademlias will be empty when retrieve requests happen
+		time.Sleep(50 * time.Millisecond)
```

Comment:
Is it possible that if this sleep is too short, the test passes but from different reason then this test is designed for? It would not have any peers in kademlia. Should there be some sort of check to see if peers are connected?

Reply:
it would fail in both cases. i added a check nevertheless

Reply:
Thanks. Unfortunately, now it timing out https://travis-ci.org/ethersphere/swarm/jobs/652344397#L1236.

Reply:
pushed a fix